### ISSUE TITLE
custom-metrics-autoscaling/direct-to-sd/custom-metrics-sd.yaml deployment failed

### ISSUE THREAD

**Issue by rick-c-goog** [GoogleCloudPlatform/kubernetes-engine-samples#385]
Tried to run this direct-to-sd custom-metric HPA example:
out of box failed: unable to start container process: exec: "./sd-dummy-exporter": stat ./sd-dummy-exporter: no such file or directory: unknown

I tried to build and push image to artifact registry, update the custom-metrics-sd.yaml with my own image, still experienced the same error.

**Comment by rick-c-goog** [comment]
Yes, I believe I got the issue resolved now. 
 File https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/custom-metrics-autoscaling/direct-to-sd/custom-metrics-sd.yaml

line  - command: ["./sd-dummy-exporter"] should be changed to:
       - command: ["./direct-to-sd"]

With changes above, the tutorial works on standard GKE. But it will fail on GKE Auto-pilot, 
even though I followed instruction for workload Identity related commands from 
https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/custom-metrics-stackdriver-adapter

logs from adaptor itself showing no errors, 
but logs from "kubectl logs po custom-metric-sd-7cb9c87cf9-p448f" shows SA permission issue while the GSA already has all the needed permission. 
Error:
Failed to write time series data for new resource model: googleapi: Error 403: Permission monitoring.timeSeries.create denied (or the resource may not exist)., forbidden 

Any idea why same example won't work on GKE Auto Pilot?
